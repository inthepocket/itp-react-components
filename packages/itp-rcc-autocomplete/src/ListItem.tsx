import * as React from 'react';
import classnames from 'classnames';
import AutoCompleteContext from './AutoCompleteContext';
import AriaInterface from './AriaInterface';

const { Consumer } = AutoCompleteContext;

const ListItem = ({
  id,
  isHighlighted,
  isSelected,
  value,
  ...ariaProps
}: ListItemPropsInterface) => (
  <Consumer>
    {({ prefixCss, styles }) => {
      const containerClassName = styles ? styles.listItem : `${prefixCss}__listItem`;
      const highlightedClassName = styles ? styles.isHighlighted : `${prefixCss}__listItem--isHighlighted`;
      const selectedClassName = styles ? styles.isSelected : `${prefixCss}__listItem--isSelected`;

      const className = classnames({
        [`${containerClassName}`]: containerClassName,
        [`${highlightedClassName}`]: isHighlighted && highlightedClassName,
        [`${selectedClassName}`]: isSelected && selectedClassName,
      });

      return (
        <li {...ariaProps} className={className}>
          {value}
        </li>
      );
    }}
  </Consumer>
);

export interface ListItemInterface {
  value: string;
}

export interface ListItemPropsInterface extends AriaInterface {
  id: string;
  isHighlighted?: boolean;
  isSelected?: boolean;
  value: string;
}

export default ListItem;
