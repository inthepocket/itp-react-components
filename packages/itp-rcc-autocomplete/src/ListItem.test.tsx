import * as React from 'react';
import { create } from 'react-test-renderer';
import ListItem from './ListItem';

test('Autocomplete ListItem snapshot', () => {
  const tree = create((
    <ListItem
      id="amsterdam"
      value="Amsterdam"
    />
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});