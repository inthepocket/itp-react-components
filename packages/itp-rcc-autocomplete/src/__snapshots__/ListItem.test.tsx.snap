// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete ListItem snapshot 1`] = `
<li
  className="card__listItem"
>
  Amsterdam
</li>
`;
